### PR TITLE
381 rest api refactor phase3

### PR DESCRIPTION
**Pull request by selimnairb**
This addresses #381.

### CODE REVIEW

File: hs_core/views/pagination.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from rest_framework.pagination import PageNumberPagination
+
+class SmallDatumPagination(PageNumberPagination):
+    """ Only use for requests whose resulting datum elements are small and where
+        one wants to force all results to be on one page
+    """
+    page_size = 10000
```
Comment (ironfroggy) [inline hs_core/views/pagination.py]: if you really want this to show ALL then i think you can set page_size to None

File: hs_core/views/resource_rest_api.py [hunk 2]
```diff
@@ -275,6 +325,84 @@ def create(self, request, *args, **kwargs):
...
+        "date_created": date resource created,
+        "date_last_updated": date resource last updated,
+        "public": True or False,
+        "bag_url": link to bag file,
+        "science_metadata_url": link to science metadata
+    }
+    """
+    @property
+    def allowed_methods(self):
+        return ['GET']
```
Comment (ironfroggy) [inline hs_core/views/resource_rest_api.py]: no reason to make this a method if its static anyway.

`allowed_methods = ('GET',)`

File: hs_core/views/resource_rest_api.py [hunk 3]
```diff
@@ -275,6 +325,84 @@ def create(self, request, *args, **kwargs):
...
+class AccessRulesUpdate(APIView):
+    """
+    Set access rules for a resource
+
+    REST URL: hsapi/resource/accessRules/{pk}
+    HTTP method: PUT
+
+    :type pk: str
+    :param pk: id of the resource
+    :return: No content.  Status code will 204 (No content)
```
Comment (ironfroggy) [inline hs_core/views/resource_rest_api.py]: Actual status is 200, and I think that's a better status code than 204 for this

File: hs_core/views/serializers.py [hunk 4]
```diff
@@ -81,6 +91,15 @@ class ResourceListItemSerializer(serializers.Serializer):
     science_metadata_url = serializers.URLField()
 
 
+class ResourceType(object):
+    def __init__(self, resource_type):
+        self._resource_type = resource_type
+
+    @property
+    def resource_type(self):
+        return self._resource_type
```
Comment (ironfroggy) [inline hs_core/views/serializers.py]: can get rid of this by just dropping the prefix underscore from `_resource_type`

**Comment by selimnairb** [comment]
I believe I have addressed all concerns identified so far.  @ironfroggy @pkdash can this be merged?

**Comment by pkdash** [comment]
+1

**Pull request merged**